armi/reactor/components/component.py

```diff
@@ -1309,6 +1315,21 @@ def getFuelMass(self) -> float:
         """Return the mass in grams if this is a fueled component."""
         return self.getMass() if self.hasFlags(flags.Flags.FUEL) else 0.0
 
+    def finalizeLoadingFromDB(self):
+        """Apply any final actions after creating the component from database.
+
+        This should **only** be called internally by the database loader. Otherwise
+        some properties could be doubly applied.
+        """
+        # This exists because the theoretical density is initially defined as a material
+        # modification, and then stored as a Material attribute. When reading from blueprints,
+        # the blueprint loader sets the theoretical density parameter from the Material
+        # attribute. Component parameters are also set when reading from the database.
+        # But, we need to set the Material attribute so routines that fetch a material's
+        # density property account for the theoretical density.
+        # See https://github.com/terrapower/armi/issues/1440
+        self.material.adjustTD(self.p.theoreticalDensityFrac)
```

Comment:
```suggestion
        some properties could be doubly applied.

        This exists because the theoretical density is initially defined as a material
        modification, and then stored as a Material attribute. When reading from blueprints,
        the blueprint loader sets the theoretical density parameter from the Material
        attribute. Component parameters are also set when reading from the database.
        But, we need to set the Material attribute so routines that fetch a material's
        density property account for the theoretical density.
        """
        self.material.adjustTD(self.p.theoreticalDensityFrac)
```

Reply:
Two things:

1. We don't link tickets or PRs in the docstrings. Either we document completely in the code, or it's not worth documenting. (At least in ARMI.)
2. I figure if we have 15 lines of comments, that could go in the docstring. Thoughts?

Reply:
> I figure if we have 15 lines of comments, that could go in the docstring. Thoughts?

🤷 I could go either way. It felt a bit too implementation-specific / in-the-weeds for a docstring. But the comment-to-code ratio is high enough that it probably deserves a bit more significance